src/main/java/org/mockito/internal/junit/ExceptionFactory.java

```diff
@@ -4,49 +4,63 @@
  */
 package org.mockito.internal.junit;
 
-import junit.framework.ComparisonFailure;
 import org.mockito.exceptions.verification.ArgumentsAreDifferent;
 
 public class ExceptionFactory {
 
-    private final static boolean hasJUnit = canLoadJunitClass();
-
     private ExceptionFactory() {
     }
 
-    /**
-     * If JUnit is used, an AssertionError is returned that extends from JUnit {@link ComparisonFailure} and hence provide a better IDE support as the comparison result is comparable
-     */
-    public static AssertionError createArgumentsAreDifferentException(String message, String wanted, String actual) {
-        if (hasJUnit) {
-            return createJUnitArgumentsAreDifferent(message, wanted, actual);
-        }
-        return new ArgumentsAreDifferent(message);
+    private static interface ExceptionFactoryImpl {
+        AssertionError create(String message, String wanted, String actual);
     }
 
-    private static AssertionError createJUnitArgumentsAreDifferent(String message, String wanted, String actual) {
-        return JUnitArgsAreDifferent.create(message, wanted, actual);
-    }
+    private final static ExceptionFactoryImpl factory;
+
+    static {
+        ExceptionFactoryImpl theFactory = null;
 
-    private static boolean canLoadJunitClass() {
         try {
-            JUnitArgsAreDifferent.create("message", "wanted", "actual");
-        } catch (Throwable onlyIfJUnitIsNotAvailable) {
-            return false;
+            // The following is neater but requires -source and -target >= 1.8
+            // theFactory = org.mockito.exceptions.verification.opentest4j.ArgumentsAreDifferent::new;
+            theFactory = new ExceptionFactoryImpl() {
+                @Override
+                public AssertionError create(String message, String wanted, String actual) {
+                    return new org.mockito.exceptions.verification.opentest4j.ArgumentsAreDifferent(message, wanted, actual);
+                }
+            };
+        } catch (Throwable onlyIfOpenTestIsNotAvailable) {
+            try {
+                // The following is neater but requires -source and -target >= 1.8
+                // theFactory = org.mockito.exceptions.verification.junit.ArgumentsAreDifferent::new;
+                theFactory = new ExceptionFactoryImpl() {
+                    @Override
+                    public AssertionError create(String message, String wanted, String actual) {
+                        return new org.mockito.exceptions.verification.junit.ArgumentsAreDifferent(message, wanted, actual);
+                    }
+                };
+            } catch (Throwable onlyIfJUnit3IsNotAvailable) {
+            }
         }
-        return true;
+        // The following is neater but requires -source and -target >= 1.8
```

Comment:
We will get to that when we switch to Java 1.8. Let's remove these comments for now.

Reply:
Done.